design-proposals/klip-43-timestamp-data-type-support.md

```diff
@@ -0,0 +1,159 @@
+# KLIP-43: TIMESTAMP Data Type Support
+
+**Author**: @jzaralim | 
+**Release Target**: 0.15 | 
+**Status**: In Discussion | 
+**Discussion**: https://github.com/confluentinc/ksql/pull/6649
+
+**tl;dr:** _KSQL currently converts time data into long values. This KLIP introduces a TIMESTAMP
+data type to handle time data instead._
+
+## Motivation and background
+
+With Connect integration, KSQL processes data from various sources and sinks. Most other databases
+have a TIMESTAMP or DATETIME type, which KSQL currently converts into long values. This sometimes
+causes problems when sinking into a database. Supporting TIMESTAMP types would make moving time data
+between KSQL and other data sources/sinks smoother and less error-prone.
```

Comment:
Let's extend a little bit what the current problems exist and the benefits users would get when TIMESTAMP column types exist. Seems useful to differentiate between TIMESTAMP column types vs data types. As you mentioned at some point, delimited and JSON will use long values as the data format type, and AVRO will use its own data format type (with ending being a long value too). 

A few comments from above:

`With Connect integration, KSQL processes data from various sources and sinks.`. This seems not in the right context. `sources and sinks` are also a concept for CREATE_AS queries, not related to Connect. What's the real problem when users use Connect and are migrating tables with timestamps? I don't know about them, so it would be good to know what's wrong. Errors during the Connect migration? Users require UDF to migrate them? What's wrong?

`which KSQL currently converts into long values.` I'm not sure about this one. As I mentioned in the TLDR section, long values are used in the final format written to the topic anyway, right? Is the intention here that ksqlDB always use long column types to reference to timestamps?

`This sometimes causes problems when sinking into a database.` What's the problem when sinking? If I store a timestamp as long value, then use UDF to read from that sink, then it should be read fine, right? Or is there an actual problem trying to store the timestamp as long value?

`Supporting TIMESTAMP types would make moving time data between KSQL and other data sources/sinks smoother  and less error-prone.` This is good. Are there any other reasons? Look at the problems I mentioned in the TLDR section.